File: language/python/modules/builtins/Classes/list/insert.py
# Mutates and returns None

# Inserts an element before the index
# All the elements after need to be shifted (copied) into a higher index
# O(n) at worst scenario (inserting at the beginning of the list)

# %%
my_list = ["a", "c"]
my_list.insert(1, "b")
my_list

File: language/c++/libraries/list/classes/list/insert.cpp
#include <iostream>
#include <list>
using namespace std;

int main() {
  list<int> myList = {1, 2, 3};

  // Insert an element at a specific position
  // The other elements are then "shifted" (not exactly because it's a LL)
  myList.insert(myList.begin(), 0);

  for (auto it = myList.begin(); it != myList.end(); it++) {
    cout << *it << endl;
  }

  return 0;
}
